xgboost-sys/build.rs

```diff
@@ -24,29 +24,27 @@ fn main() {
     // CMake
     #[cfg(feature = "cuda")]
     let dst = Config::new(&xgb_root)
-        .uses_cxx11()
         .define("BUILD_STATIC_LIB", "ON")
         .define("USE_CUDA", "ON")
         .define("BUILD_WITH_CUDA", "ON")
-        .define("BUILD_WITH_CUDA_CUB", "ON");
+        .define("BUILD_WITH_CUDA_CUB", "ON")
+        .define("CMAKE_CXX_STANDARD", "17");
 
     #[cfg(not(feature = "cuda"))]
     let mut dst = Config::new(&xgb_root);
```

Comment:
Would it be cleaner to move this up above
```
     #[cfg(feature = "cuda")]
```

so that we're only creating the Config::new in one place, and then optionally doing the cuda definitions?